aptos-move/framework/aptos-framework/sources/randomness.move

```diff
@@ -0,0 +1,23 @@
+/// On-chain randomness utils.
+module aptos_std::randomness {
+    use aptos_framework::system_addresses;
+
+    friend aptos_framework::block;
+
+    /// The block-level seed randomness.
+    /// It's updated at the beginning of every block.
+    struct BlockRandomness has drop, key {
+        block_randomness: vector<u8>,
+    }
+
+    /// Invoked in `block_prologue_ext()` to update the block-level seed randomness.
+    public(friend) fun on_new_block(account: &signer, randomness_available: bool, block_randomness: vector<u8>) acquires BlockRandomness {
```

Comment:
I'd have a separate transaction to initialize this and never use move_to here to update randomness_available